### PR TITLE
Add support for pypy 2

### PR DESCRIPTION
**Pull request by michaeladda**
Tested by creating layers for both pypy2 and pypy3.5.

Also manually tested the sls example using pypy2 runs well.

### CODE REVIEW

File: bootstrap.py2 [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+
+def init_error(message, type):
+    details = {"errorMessage": message, "errorType": type}
+    details = json.dumps(details).encode("utf-8")
+    req = request.Request(
+        "http://{}/2018-06-01/runtime/init/error".format(RUNTIME_API),
+        details,
+        {"Content-Type": "application/json"},
+    )
+    res = request.urlopen(req)
```
Comment (vogre) [inline bootstrap.py2]: In `diff bootstrap.py3 bootstrap.py2`

```
-    with request.urlopen(req) as res:
-        res.read()
+    res = request.urlopen(req)
+    res.read()
```

Looks like this change doesn't close the response, and the recommended way seems to be `contextlib.closing` https://docs.python.org/2/library/contextlib.html.

File: conf.sh [hunk 2]
```diff
@@ -0,0 +1,18 @@
+PYPY_REGIONS=(
+  #ap-northeast-1
+  #ap-northeast-2
+  #ap-south-1
+  #ap-southeast-1
+  #ap-southeast-2
+  #ca-central-1
+  eu-central-1
```
Comment (vogre) [inline conf.sh:8]: Do we need to re-enable other regions for publish?

File: examples/sam/template.yml [hunk 3]
```diff
@@ -1,14 +1,14 @@
 AWSTemplateFormatVersion: '2010-09-09'
 Transform: AWS::Serverless-2016-10-31
-Description: "IOPipe PyPy 3.5 Runtime SAM Example"
+Description: "PyPy Runtime SAM Example"
 
 Globals:
   Function:
     Layers:
-        - arn:aws:lambda:us-east-1:146318645305:layer:pypy35:1
+        - arn:aws:lambda:us-east-1:146318645305:layer:pypy35600:1
```
Comment (vogre) [inline examples/sam/template.yml]: Where does this ARN come from?

File: create-buckets.sh [hunk 4]
```diff
@@ -1,11 +1,11 @@
 #!/bin/bash -e
```
Comment (vogre) [inline create-buckets.sh]: `-u` would be safe also here?

File: latest-layer-arns.sh [hunk 5]
```diff
@@ -1,8 +1,8 @@
 #!/bin/bash -e
```
Comment (vogre) [inline latest-layer-arns.sh]: `-u` should be safe here too (no undefined variables)?

File: Makefile [hunk 6]
```diff
@@ -1,18 +1,38 @@
 SHELL=/bin/bash
 
-all: clean pypy35.zip upload publish
+PYPY_VERSIONS := pypy-7.1.1 pypy3.5-6.0.0
```
Comment (vogre) [inline Makefile:3]: This is following the convention of `squeaky/protable-pypy`, right?

**Review by vogre — COMMENTED** [review]
Thanks!

Note that @arielshaqed had some Makefile comments.

File: Makefile (hunk 6 above)
Comment (michaeladda) [reply Makefile:3]: Indeed.

File: bootstrap.py2 (hunk 1 above)
Comment (michaeladda) [reply bootstrap.py2]: Thanks. Fixed. PTAL

File: conf.sh (hunk 2 above)
Comment (michaeladda) [reply conf.sh:8]: Not unless we intend to release the layer in a public manner world wide
(which we don't atm)

File: examples/sam/template.yml (hunk 3 above)
Comment (michaeladda) [reply examples/sam/template.yml]: It's the iopipe account id i suspect
(which is the original repo this project was forked from)

**Review by michaeladda — COMMENTED** [review]
Thanks for the review. Addressed and re-tested. Please take another look.

File: bootstrap.py2 [hunk 7]
```diff
@@ -0,0 +1,137 @@
...
+    return request_id, event, context
+
+
+def invocation_response(request_id, handler_response):
+    if not isinstance(handler_response, (bytes, str)):
+        handler_response = json.dumps(handler_response, default=decimal_serializer)
+    if not isinstance(handler_response, bytes):
+        handler_response = handler_response.encode("utf-8")
+    req = request.Request(
+        "http://{}/2018-06-01/runtime/invocation/{}/response".format(RUNTIME_API, request_id),
```
Comment (vogre) [inline bootstrap.py2]: Is there a reason to change `%` format to `{}` format? Asking only for minimal diff between bootstrap versions, currently this is the biggest change, although quick to review.

**Review by vogre — APPROVED** [review]
Thanks!

**Review by vogre — APPROVED** [review]
Awesome, thanks!

**Pull request merged**